jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/popup/CodeWhispererPopupManager.kt

```diff
@@ -300,6 +300,12 @@ class CodeWhispererPopupManager {
                 states.responseContext,
                 perceivedLatency
             )
+            states.requestContext.latencyContext.codewhispererPostprocessingEnd = System.nanoTime()
+            states.requestContext.latencyContext.codewhispererEndToEndEnd = System.nanoTime()
+
+            if (!CodeWhispererInvocationStatus.getInstance().hasExistingInvocation()) {
+                CodeWhispererTelemetryService.getInstance().sendClientComponentLatencyEvent(states)
+            }
```

Comment:
Why is this event duplicated in CodeWhispererUIChangeListener.kt? Are there two different entrypoints that never join? Emitting stateful events at multiple points in the code could cause issues later as the codebase evolves.

Reply:
These two will never join although it's nontrivial to realize it for just reviewing the PR. Agreeing that a single emitting point makes the most sense to be less bug-prone in the future.

Reply:
The calling point here corresponds to this [comment](https://github.com/aws/aws-toolkit-jetbrains/pull/3437#discussion_r1065160422), for path 1)

Reply:
I can do a small refactor to make it to have a single emitting point.

Reply:
If it's a small change then yeah I'd go for it. Otherwise don't worry about it.